api/controller.py

```diff
@@ -727,18 +729,30 @@ def search(self, lane_identifier):
             # Send the search form
             return OpenSearchDocument.for_lane(lane, this_url)
 
+        language_header = flask.request.headers.get("Accept-Language")
+        if language_header:
+            languages = parse_header(language_header)
+            languages = map(lambda x: str(x), languages)
+        else:
+            languages = None
+
         pagination = load_pagination_from_request(default_size=Pagination.DEFAULT_SEARCH_SIZE)
         if isinstance(pagination, ProblemDetail):
             return pagination
 
         # Run a search.
-        this_url += "?q=" + urllib.quote(query.encode("utf8"))
+        if media:
+            media_url = "&media=" + urllib.quote(media.encode("utf8"))
+        else:
+            media_url = ''
```

Comment:
I'm on the fence about whether, having loaded the initial set of languages from the request, that language set should be propagated to future pages through the search URL. Right now if someone loads page 1 of search results, then switches their language settings and follows the 'next' link, they will get the second page search results from their new language settings, rather than the second page of the search they originally ran.

In real life this difference doesn't matter, so I wouldn't put too much work into fixing this, but in general I think it's important to make the 'next' link explicitly propagate all the values that went into the original query, even values that didn't originally come from the query string.

Reply:
Updated.